part-iii-infrastructure-components/event-processing.md

```diff
@@ -209,3 +209,37 @@ Besides these provided policies, you can define your own. All policies must impl
 
 It is recommended to explicitly define an `ErrorHandler` when using the `AsynchronousEventProcessingStrategy`. The default `ErrorHandler` propagates exceptions, but in an asynchronous execution, there is nothing to propagate to, other than the Executor. This may result in Events not being processed. Instead, it is recommended to use an `ErrorHandler` that reports errors and allows processing to continue. The `ErrorHandler` is configured on the constructor of the `SubscribingEventProcessor`, where the `EventProcessingStrategy` is also provided.
 
+### Replaying events
+
+In cases when you want to rebuild projections (view models), replaying past events comes in handy. The idea is to start from the beginning of time and invoke all event handlers anew. The `TrackingEventProcessor` supports replaying of events. In order to achieve that, you should invoke the `resetTokens()` method on it (it's important to know that the Processor must not be in active state, so it's wise to shut it down first, reset it and then start it up again). It is possible to define a `@ResetHandler`, so you can do some preparation prior to resetting. Let's take a look how we can accomplish replaying. First, we'll see one simple projecting class:
+
+```java
+@ProcessingGroup("projections")
+public class MyProjection {
+    ...
+    @EventHandler
+    public void on(MyEvent evt, ReplayStatus replayStatus) { // we can wire a ReplayStatus here so we can see whether this
+                                                             // event is delivered to our handler as a 'REGULAR' event or
+                                                             // 'REPLAY' event
+        // do event handling
+    }
+    
+    @ResetHandler
+    public void onReset() { // will be called before replay starts
+        // do pre-reset logic
```

Comment:
Maybe expand this comment to `// Do pre-reset logic, like clearing out the Projection table for a clean slate.` to give an example why you'd want a `@ResetHandler` annotated function?